core/document/src/layers/ellipse.rs

```diff
@@ -0,0 +1,34 @@
+use super::style;
+use super::LayerData;
+
+use std::fmt::Write;
+
+#[derive(Debug, Clone, Copy, PartialEq, Default)]
+pub struct Ellipse {
+	shape: kurbo::Ellipse,
+	style: style::PathStyle,
+}
+
+impl Ellipse {
+	pub fn new(center: impl Into<kurbo::Point>, radii: impl Into<kurbo::Vec2>, rotation: f64, style: style::PathStyle) -> Ellipse {
+		Ellipse {
+			shape: kurbo::Ellipse::new(center, radii, rotation),
+			style,
+		}
+	}
+}
+
+impl LayerData for Ellipse {
+	fn render(&mut self, svg: &mut String) {
+		let _ = write!(
+			svg,
+			r#"<ellipse cx="0" cy="0" rx="{}" ry="{}" transform="translate({} {}) rotate({})" {} />"#,
+			self.shape.radii().x,
+			self.shape.radii().y,
+			self.shape.center().x,
+			self.shape.center().y,
+			self.shape.rotation().to_degrees(),
+			self.style.render(),
```

Comment:
This would be a lot more readable with named format arguments:
```
… rx="{rx}" ry="{ry}" transform="translate({cx} {cy}) rotate({rot})" {}/>"#, rx=self.shape.radii().x, ry=self.shape.radii().y, …
```
also the multiple calls to `radii` and `center` should be factored out. You can even do it like this:
```
let kurbo::Vec2 { x: rx, y: ry } = self.shape.radii();
let kurbo::Point { x: cx, y: cy } = self.shape.center();
```
note that format args don't have shorthand syntax afaik, so you still have to write `cx=cx` etc.

Reply:
It looks like format arguments requires the unstable feature `format_args_capture`. I'll factor out the `radii`and `center` calls but leave the arguments for now.